api-core/src/main/java/com/vimeo/networking2/internal/interceptor/AcceptHeaderInterceptor.kt

```diff
@@ -32,11 +32,15 @@ import okhttp3.Response
  */
 class AcceptHeaderInterceptor : Interceptor {
     override fun intercept(chain: Interceptor.Chain): Response = chain.proceed(
-        chain.request().newBuilder().header(HEADER_ACCEPT, HEADER_ACCEPT_VALUE).build())
+        chain.request().newBuilder().header(
+            HEADER_ACCEPT,
+            HEADER_ACCEPT_VALUE
+        ).build()
+    )
 
     companion object {
         private const val HEADER_ACCEPT = "Accept"
         private const val HEADER_ACCEPT_VALUE: String =
-                "application/vnd.vimeo.*+json; version=${ApiConstants.API_VERSION}"
+            "application/vnd.vimeo.*+json; version=${ApiConstants.API_VERSION}"
```

Comment:
The other headers that are set are fixed for the duration of of the lifecycle of a client instance. Originally when we were designing the new client, we thought that it would be better to set the authorization header in an interceptor. However, when building the new client, it didn't seem very practical. 

The reason is that an interceptor is injected into an OkHttp instance, which is injected into a Retrofit instance. This would create coupling between the interceptor instance and both the OkHttp instance and the `VimeoApiClient` instance. The relationships between the classes get a little complicated, and maintaining the proper states becomes difficult. The interceptor either needs to be mutable, rely on an `AccountStore`, or rely on the `Authenticator` instance. Making the interceptor mutable feels bad, but we can't use an `AccountStore` unless it is the same one used by the `Authenticator`. We can't use the `Authenticator` because it isn't created when we construct the interceptor. Overall, it seemed more straight forward to manage the auth state within the `Authenticator` and pass the header manually within the `VimeoApiClient`. We also avoid the problem of needing to use different auth headers for `VimeoApiClient` and `Authenticator`.

Reply:
That makes a lot of sense. Thanks for the clear explanation.